bikeshack/smd
# Create subroutine to parse data in JSON format and populate database

## Summary and Scope

This PR adds the `parseRedfishPostData` function in `smd-api.go` to parse data sent from [Magellan](https://github.com/bikeshack/magellan) in a JSON format and store `ComponentEndpoint`'s and `Component`'s in the postgres database for `hms-bss`. This feature is optional and is only used when the appropriate value is stored in the HTTP when making a HTTP request.

This change is backwards compatible with previous versions of SMD since it only adds an extra step to populating the database without changing how anything else works. Therefore, the original behavior of SMD is still maintained.

## Issues and Related PRs

There are no issues or other related PRs.

## Testing

_List the environments in which these changes were tested._

Tested on:

- WSL Ubuntu 22.04 environment
- Actual cluster of hardware (SI)

Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Start up postgres Docker container
- Run SMD instance
- Run `magellan scan` then a `magllan collect` (or just a HTTP request with similar output to magellan)
- Query the `../hsm/v2/State/Components` and `../hsm/v2/Inventory/ComponentEndpoints` to confirm that it's work (should return output for both)

## Risks and Mitigations

There are no known issues with this PR.


## Pull Request Checklist
